### PR TITLE
✨ Add platform PURLs

### PR DESCRIPTION
**Pull request by czunker**
Add the asset platform in package URL format.

### CODE REVIEW

File: providers/os/resources/purl/platform_purl_test.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+			want:    "pkg:os/esxi/@7.0.3?arch=x86_64&distro=esxi-7.0.3",
+			wantErr: "",
+		},
+		{
+			name: "kubernetes deployment",
+			platform: &inventory.Platform{
+				Name:    "k8s-deployment",
+				Version: "1.27",
+			},
+			want:    "pkg:os/k8s-deployment/@1.27?distro=k8s-deployment-1.27",
```
Comment (czunker) [inline providers/os/resources/purl/platform_purl_test.go]: This looks wrong. It is not an os and not a distro.

File: providers/os/resources/purl/platform_purl_test.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			wantErr: "",
+		},
+		{
+			name: "aws platform",
+			platform: &inventory.Platform{
+				Name: "aws",
+			},
+			want:    "pkg:os/aws/?distro=aws",
```
Comment (czunker) [inline providers/os/resources/purl/platform_purl_test.go]: Same for cloud providers.

Other terms I could think of:
- platform
- runtime

???

Comment (chris-rock) [reply providers/os/resources/purl/platform_purl_test.go]: Let's switch to platform

**Review by chris-rock — APPROVED** [review]
Nice, lets try it

**Pull request merged**